### ISSUE TITLE
Possibility to add custom palettes

### ISSUE THREAD

**Issue by joewdavies** [riatelab/dicopal.js#8]
Bonjour!

In our app we combine custom colour palettes with Dicopal. I was wondering if there is a method available to add palettes to Dicopal, or if it could be added if it isn't too much trouble? 

For example:

```
let myPalette = {
  id: "myCustomPalettes_6",
  name: "myCustomPalettes",
  number: 6,
  type: "sequential",
  colors: ["#ffeb99","#c7e9b4","#7fcdbb","#41b6c4","#104f99","#17256b"],
  provider: "Eurostat",
  url: "https://github.com/myPalette"
}
dicopal.addPalette(myPalette)
```

That way we could let dicopal handle all of the colour palette logic in our app!

Cheers,
Joe

**Comment by mthh** [comment]
Hi!

It's a great idea. I'll have to see what's involved and if I don't see anything too blocking I think I'll implement such a feature :+1:

**Comment by mthh** [comment]
Little progress has been made on this issue but it's still on my todo list for the next few days :sweat: 

One of the main "problems" I saw is that to work correctly on user-added palettes (i.e. to handle the different possible cases, depending on whether the user requests with or without a central class) `getAsymmetricDivergingColors` would require the user to have provided at least one variation of its diverging palette with a central class).

I think I can manage the case where the user does not provide a version with a central class, either with appropriate error messages thrown by `getAsymmetricDivergingColors`, and/or by proposing to pass `getAsymmetricDivergingColors` a color for the central class (rather than just true / false).
I'll probably go with the latter idea because our application which uses dicopal is likely to need this kind of functionality (creation of asymmetrical divergent palettes + possibility of overwriting the value of the central class with a value of the user's choice).

**Comment by joewdavies** [comment]
Sounds good to me!

No worries, whenever you can 🙂

Once again thanks for your work!

**Comment by mthh** [comment]
Released in 0.8.0 :tada:

**Comment by joewdavies** [comment]
Great! Will give it a go after easter, thanks! 😄
